### PR TITLE
feat(condo): DOMA-5613 create PredictTransactionClassificationService

### PR DESCRIPTION
**Pull request by VKislov**
In tests it is not possible to check the flow, therefore, as a proof of work, I copied the request from the code to postman and received the correct answer
image

### CODE REVIEW

File: apps/condo/domains/banking/schema/PredictTransactionClassificationService.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        {
+            access: true,
+            type: 'type PredictTransactionClassificationOutput { id: ID!, name: String!, isOutcome: Boolean!, category: Category }',
+        },
+    ],
+    
+    queries: [
+        {
+            access: access.canPredictTransactionClassification,
+            schema: 'predictTransactionClassification (data: PredictTransactionClassificationInput!): PredictTransactionClassificationOutput',
```
Comment (AntonAL) [inline apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:64]: Could you please add docs to service.
You can also add `errors` into `doc` object:

```
doc: {
    summary: '…',
    description: '…',
    errors,
},
```

To see how it works, take a look to `schemaDocPreprocessor`.

File: apps/condo/domains/banking/schema/PredictTransactionClassificationService.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+const { NOT_FOUND } = require('@condo/domains/common/constants/errors')
+
+const ML_SPACE_TRANSACTION_CLASSIFIER = conf['ML_SPACE_TRANSACTION_CLASSIFIER'] ? JSON.parse(conf['ML_SPACE_TRANSACTION_CLASSIFIER']) : {}
+
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const ERRORS = {
```
Comment (AntonAL) [inline apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:21]:
```suggestion
const errors = {
```

File: apps/condo/domains/banking/schema/PredictTransactionClassificationService.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+                const { endpoint, authKey, workspace } = ML_SPACE_TRANSACTION_CLASSIFIER
+                if (!endpoint || !authKey || !workspace) {
+                    throw new GQLError(ERRORS.ML_SPACE_NOT_CONFIGURED, context)
+                }
+                const response = await fetch(endpoint, {
+                    headers: {
+                        'Accept': 'application/json',
+                        'Content-Type': 'application/json',
+                        'cookie': `authservice_session=${authKey}`,
```
Comment (AntonAL) [inline apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:79]: I see that this header is used in `PredictTicketClassificationService`, but where exactly this header will be handled?

Comment (VKislov) [reply apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:84]: condo-classifier-api

File: apps/condo/domains/banking/schema/PredictTransactionClassificationService.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                const { data: { purpose } } = args
+                if (conf.NODE_ENV === 'test' || !ML_SPACE_TRANSACTION_CLASSIFIER) {
+                    return null
+                }
+
+                const { endpoint, authKey, workspace } = ML_SPACE_TRANSACTION_CLASSIFIER
+                if (!endpoint || !authKey || !workspace) {
+                    throw new GQLError(ERRORS.ML_SPACE_NOT_CONFIGURED, context)
+                }
+                const response = await fetch(endpoint, {
```
Comment (AntonAL) [inline apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:80]: Maybe to extract fetching logic into a class?
This way all the headers and response handling logic will not be duplicated across tickets and banking.

**Review by AntonAL — COMMENTED** [review]
You can mock `fetch` to test it

File: apps/condo/domains/banking/schema/PredictTransactionClassificationService.js (hunk 4 above)
Comment (VKislov) [reply apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:80]: can be later realized as technical debt. It seems that now there are enough tasks for our feature, but there is not enough time :)

File: apps/condo/domains/banking/schema/PredictTransactionClassificationService.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    ],
+    
+    queries: [
+        {
+            access: access.canPredictTransactionClassification,
+            schema: 'predictTransactionClassification (data: PredictTransactionClassificationInput!): PredictTransactionClassificationOutput',
+            doc: {
+                summary: 'Classifier transaction by payment purpose',
+                description: 'A tool to make the customer journey easier. Allows you to classify a transaction without user intervention. Referring to the condo-classifier-api.',
+                errors: errors,
```
Comment (AntonAL) [inline apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:68]:
```suggestion
                errors,
```

File: apps/condo/domains/banking/schema/PredictTransactionClassificationService.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        },
+    ],
+    
+    queries: [
+        {
+            access: access.canPredictTransactionClassification,
+            schema: 'predictTransactionClassification (data: PredictTransactionClassificationInput!): PredictTransactionClassificationOutput',
+            doc: {
+                summary: 'Classifier transaction by payment purpose',
+                description: 'A tool to make the customer journey easier. Allows you to classify a transaction without user intervention. Referring to the condo-classifier-api.',
```
Comment (AntonAL) [inline apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:67]:
```suggestion
                description: 'Matching is performed by empirical model, implemented in external microservice "condo-classifier-api"',
```

File: apps/condo/domains/banking/schema/PredictTransactionClassificationService.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            type: 'type PredictTransactionClassificationOutput { id: ID!, name: String!, isOutcome: Boolean!, category: Category }',
+        },
+    ],
+    
+    queries: [
+        {
+            access: access.canPredictTransactionClassification,
+            schema: 'predictTransactionClassification (data: PredictTransactionClassificationInput!): PredictTransactionClassificationOutput',
+            doc: {
+                summary: 'Classifier transaction by payment purpose',
```
Comment (AntonAL) [inline apps/condo/domains/banking/schema/PredictTransactionClassificationService.js:66]: Please, specify what it does by fact

```suggestion
                summary: 'Returns id of BankCostItem corresponding to specified payment purpose string',
```

**Pull request merged**